### ISSUE TITLE
Incorrect positions of bounding boxes

### ISSUE THREAD

**Issue by shreymohan** [mystic123/tensorflow-yolo-v3#24]
On running the code demo.py, it is giving inaccurate position of bounding boxes although the confidence is quite high.
![result](https://user-images.githubusercontent.com/24687852/45909342-2e1ab580-bdb6-11e8-9997-ac9e31039986.jpg)

This is an image from coco-validation-2017 dataset. A similar trend is seen in many other pictures of traffic on street,etc.

Do you know what might be the problem?

**Comment by Jeremy26** [comment]
Hello,

Did you just clone repo and run on an image ?

**Comment by shreymohan** [comment]
Yes, I just got rid of a couple of bugs and it ran. Do you have any idea what might cause the above problem??

**Comment by mystic123** [comment]
It looks like problems with invalid coordinates translation during resizing. Let me explain:
Detector's input size is 416x416 px, which means that input images need to be resized to this size. Detector finds objects. To show them correctly on original image, all bounding boxe's coordinates need to be translated. I think that something was wrong in that phase.

**Comment by shreymohan** [comment]
Thank you! Not resizing the boxes according to the original image was the problem. I have another question though: In my understanding of yolo, the darknet is pre-trained on ImageNet dataset. Are you not loading these weights?

**Comment by mystic123** [comment]
File with weights exported by YOLOv3 authors has all of them, including Darknet-53's weights.

**Comment by shreymohan** [comment]
`def draw_boxes(boxes, img, cls_names, detection_size):
    draw = ImageDraw.Draw(img)
    print(np.array(img.size))
    print(np.array(detection_size))

    for cls, bboxs in boxes.items():
        color = tuple(np.random.randint(0, 256, 3))
        for box, score in bboxs:
            box = convert_to_original_size(box, np.array(detection_size), np.array(img.size))
            draw.rectangle(box, outline=color)
            draw.text(box[:2], '{} {:.2f}%'.format(cls_names[cls], score * 100), fill=color)


def convert_to_original_size(box, size, original_size):
    x_scale=original_size[0]/size[0]
    y_scale=original_size[1]/size[1]
    print(x_scale,y_scale)
    x_upper_right=int(np.round(box[0]*x_scale))
    y_upper_right=int(np.round(box[1]*y_scale))
    x_lower_left=int(np.round(box[2]*x_scale))
    y_lower_left=int(np.round(box[3]*y_scale))
    return [x_upper_right,y_upper_right,x_lower_left,y_lower_left]`

I have changed this method convert_to_original_size to resize the bounding boxes according to the original image but it is still giving incorrect positions. Do you know what might be wrong here?

**Comment by shreymohan** [comment]
solved the problem, just need to cast these lines into float : 
x_scale=original_size[0]/size[0]
y_scale=original_size[1]/size[1]

Thanks!! 
It would also help if you could commit the training block of this code as well.
